### ISSUE TITLE
Content Creator: Image Input Plugin

### ISSUE THREAD

**Issue by scottnath** [punchcard-cms/punchcard#155]
Need a no-frills way to upload images.

```gherkin
Feature: Images as content
  As a content creator
  I want add an image or images as content
  So that I can have images as part of my content

  Scenario: Upload an image
    Given that my content type requires me to add an image as data
    When I am entering content in said type
    Then I need to be able to include an image as content
      And the image selection should match the flow and style of other data points
      And I should be able to select an image to upload from my device
      And I should know what image I have selected after selection
      And this image should be stored as content after I submit the form.

  Scenario: Limit Image File Type
    Given that my content type requires me to add an image as data
      And that the requirement may be limited to specific image types
    When I am entering content in said type
    Then I should be limited to only allow selection of the image filetype or filetypes specified by said content-type.
      And I should not be able to select images which are not that filetype or filetypes

  Scenario: Upload one image
    Given that my content type limits me to adding only one image as data
     When I am filling in content with a limit to only one image
     Then the image upload input should limit selection to only one image

  Scenario: Upload multiple images
    Given that my content type requires me to add multiple images as data
      And that a images upload input should allow for multiple images to be selected
     When I am filling in content with a requirement for multiple images on one input
     Then the images upload input should allow selection of multiple images
```

---

**Images as Content**
- [ ] upload image
- [ ] file type
- [ ] only one
- [ ] a billion images

**Comment by joshkimmell** [comment]
Marking as blocked as we're waiting on money for file storage

**Comment by joshkimmell** [comment]
need external storage to unblock

**Comment by joshkimmell** [comment]
pluggable way to save files
need fs write file...

@scottnath will write a new story

**Comment by Snugug** [comment]
#242 and #243 should probably both be scenarios of the file input plugin as global configuration options

**Comment by scottnath** [comment]
@Snugug I've updated this story. I am not integrating #242/#243 as those should be stories for the `implementor`

**Comment by joshkimmell** [comment]
superseded by #195
